File: mobile/lib/services/csgames.api.dart
import 'package:CSGamesApp/utils/environment.dart';

class CSGamesApi {
    final String _apiUrl = Environment.apiUrl;
    final String _path;

    CSGamesApi(this._path);

    String url([route = ""]) {
        return '$_apiUrl/$_path/$route';
    }
}

File: dashboard/src/app/api/csgames.api.ts
import { environment } from "../../environments/environment";

export class CSGamesApi {
    private readonly API_URL = environment.CSGAMES_API_URL;
    private readonly path: string;

    constructor(path: string) {
        this.path = path;
    }

    protected url(route = ""): string {
        if (!route) {
            return `${this.API_URL}/${this.path}`;
        }
        return `${this.API_URL}/${this.path}/${route}`;
    }
}
